### ISSUE TITLE
Indent rule fighting with IntelliJ formatter for multiline where clause

### ISSUE THREAD

**Issue by vatbub** [pinterest/ktlint#2175]
## Expected Behavior
Consider the following code:
```kotlin
fun <TFeature, TValidated> applyToAllCloseFeaturesWithUiFlow(
    thisFeature: TFeature,
    allFeaturesOfThisKind: List<TFeature>,
    optionsToApply: TValidated,
// .. more parameters
) where TFeature : Clusterable,
        TFeature : SupportsExternalObjectCoordinates<out Options<out Options.Validated>, out Options.Validated, *>,
        TValidated : Options.Validated {
    // do something
}
```

IntelliJ formats the `where` clause such that the type parameters are exactly beneath each other (using 8 spaces in front of `TFeature` and `TValidated`.

## Observed Behavior
Ktlint uses 6 instead of 8 spaces and therefore fights with IntelliJ:

```kotlin
fun <TFeature, TValidated> applyToAllCloseFeaturesWithUiFlow(
    thisFeature: TFeature,
    allFeaturesOfThisKind: List<TFeature>,
    optionsToApply: TValidated,
// .. more parameters
) where TFeature : Clusterable,
      TFeature : SupportsExternalObjectCoordinates<out Options<out Options.Validated>, out Options.Validated, *>,
      TValidated : Options.Validated {
    // do something
}
```

## Steps to Reproduce
Run ktlint with the configuration and `.editorconfig` specified beneath.

## Your Environment
* Version of ktlint used: 0.50.0
* Relevant parts of the `.editorconfig` settings: See below
* Name and version (or code for custom task) of integration used: Maven through the antrun plugin (config see below)
* Version of Maven used: 3.9.3
* Operating System and version: Windows 10

Maven config of ktlint:
```xml
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-antrun-plugin</artifactId>
    <version>3.1.0</version>
    <executions>
        <execution>
            <id>ktlint</id>
            <configuration>
                <target name="ktlint">
                    <java taskname="ktlint" dir="${basedir}" fork="true" failonerror="true"
                          classname="com.pinterest.ktlint.Main" classpathref="maven.plugin.classpath">
                        <arg value="/**/src/**/*.kt"/>
                    </java>
                </target>
            </configuration>
            <goals>
                <goal>run</goal>
            </goals>
        </execution>
        <execution>
            <id>ktlint-format</id>
            <configuration>
                <target name="ktlint">
                    <java taskname="ktlint" dir="${basedir}" fork="true" failonerror="true"
                          classname="com.pinterest.ktlint.Main" classpathref="maven.plugin.classpath">
                        <arg value="-F"/>
                        <arg value="/**/src/**/*.kt"/>
                        <jvmarg value="--add-opens"/>
                        <jvmarg value="java.base/java.lang=ALL-UNNAMED"/>
                    </java>
                </target>
            </configuration>
            <goals>
                <goal>run</goal>
            </goals>
        </execution>
    </executions>
    <dependencies>
        <dependency>
            <groupId>com.pinterest</groupId>
            <artifactId>ktlint</artifactId>
            <version>0.50.0</version>
        </dependency>
        <!-- additional 3rd party ruleset(s) can be specified here -->
    </dependencies>
</plugin>
```

Relevant parts of `.editorconfig`:
```
indent_size = 4
indent_style = space
tab_width = 4
ij_continuation_indent_size = 4
```

**Comment by paul-dingemans** [comment]
Ktlint only had special formatting for the `where` keyword when used with a `class`. I have now added a similar exception for usage in `fun`. Are you aware of other construct in which `where` can be used?

**Comment by vatbub** [comment]
Thanks for such a quick fix :) I just checked and classes and functions are the only places where generics and the `where` keyword can appear (c.f. [Kotlin docs on generics](https://kotlinlang.org/docs/generics.html) ). So, everything should be covered now.
